ewatercycle/analysis/__init__.py

```diff
@@ -0,0 +1,136 @@
+import pandas as pd
+import os
+from typing import Union
+from hydrostats import metrics
+import matplotlib.pyplot as plt
+from matplotlib.dates import DateFormatter
+
+
+def hydrograph(
+    discharge: pd.DataFrame,
+    *,
+    reference: str,
+    precipitation: pd.DataFrame = None,
+    fname: Union[os.PathLike, str] = None,
+    dpi: int = None,
```

Comment:
Perhaps consider passsing **matplotlib_kwargs?

Reply:
Yeah, I was considering that. There are 2 different plots in one, the line plot (discharge) and the bar plot (observations), so this may end up being somewhat/half-complete implementation.

Did you have any specific use cases in mind?

Reply:
e.g. setting the colors, figsize, transparency